src/rgw/rgw_auth_s3.cc

```diff
@@ -501,15 +501,22 @@ static inline std::string aws4_uri_recode(const boost::string_view& src)
 
 std::string get_v4_canonical_qs(const req_info& info, const bool using_qs)
 {
-  if (info.request_params.empty()) {
+  const std::string *params = &info.request_params;
+  std::string copy_params;
+  if (params->empty()) {
     /* Optimize the typical flow. */
     return std::string();
   }
+  if (params->find_first_of('+') != std::string::npos) {
+    copy_params = *params;
+    boost::replace_all(copy_params, "+", " ");
```

Comment:
ack

Reply:
If the params begin with space or end with space, these spaces will be trimed in parse_key_value. 403 error still exists. Maybe we should replace "+" with "%20".